packages/core/src/interpreter.ts

```diff
@@ -133,11 +136,16 @@ export class Interpreter<
       ...options
     };
 
-    const { clock, logger, parent, id } = resolvedOptions;
+    const { clock, logger, parent, id, key } = resolvedOptions;
     const self = this;
 
-    // TODO: this should come from a "system"
-    this.sessionId = registry.bookId();
+    this.system = parent?.system ?? createSystem();
+
+    if (key) {
+      this.system.set(key, this);
```

Comment:
This should definitely be called in the `start` method. Without it `@xstate/react` might break when it performs its Fast Refresh.

I'm not entirely sure if we should remove this call from here though. What use cases would break without this call in the constructor?